tests/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Source.Tests/DiagnosticAnalysisRunnerTests.cs

```diff
@@ -0,0 +1,140 @@
+            {
+                var ws = new AdhocWorkspace();
+                var name = Path.GetFileNameWithoutExtension(TestProjectPath)!;
+                return ws.AddProject(ProjectInfo.Create(ProjectId.CreateNewId(), VersionStamp.Default, name, name, "C#", filePath: TestProjectPath));
+            });
+
+            var context = mock.Mock<IUpgradeContext>();
+            context.Setup(c => c.Projects).Returns(new[] { project.Object });
+            var diagnosticAnalysisRunner = new DiagnosticAnalysisRunner(GetAnalyzers(false), AdditionalTexts, loggerMock.Object);
+
+            // Act
+            var diagnostics = await diagnosticAnalysisRunner.GetDiagnosticsAsync(project.Object, CancellationToken.None).ConfigureAwait(false);
+
+            // Assert
+            Assert.Empty(diagnostics);
+        }
+
+        [Theory]
+        [InlineData(false, 0)]
+        [InlineData(true, 3)]
+        public async Task DiagnosticsWithProjectFilesTest(bool includeLocation, int expected)
+        {
+            // Arrange
+            using var mock = AutoMock.GetLoose();
+            var loggerMock = new Mock<ILogger<DiagnosticAnalysisRunner>>();
+
+            var project = CreateProject(mock);
+            project.Setup(p => p.GetRoslynProject()).Returns(() =>
+            {
+                var ws = new AdhocWorkspace();
+                var name = Path.GetFileNameWithoutExtension(TestProjectPath)!;
+                var projectInfo = ProjectInfo.Create(ProjectId.CreateNewId(), VersionStamp.Default, name, name, "C#", filePath: TestProjectPath);
+                var documents = new List<DocumentInfo>() { DocumentInfo.Create(DocumentId.CreateNewId(projectInfo.Id), "Foo.cs").WithFilePath(@"assets\TestClasses\Foo.cs") };
+                var proj = ws.AddProject(projectInfo.WithDocuments(documents));
+                return proj;
+            });
+
+            var context = mock.Mock<IUpgradeContext>();
+            context.Setup(c => c.Projects).Returns(new[] { project.Object });
+            var diagnosticAnalysisRunner = new DiagnosticAnalysisRunner(GetAnalyzers(includeLocation), AdditionalTexts, loggerMock.Object);
+
+            // Act
+            var diagnostics = await diagnosticAnalysisRunner.GetDiagnosticsAsync(project.Object, CancellationToken.None).ConfigureAwait(false);
+
+            // Assert
+            Assert.True(diagnostics.Count() == expected);
+        }
+
+        private Mock<IProject> CreateProject(AutoMock mock)
+        {
+            var project = mock.Mock<IProject>();
+            var nugetReferences = mock.Mock<INuGetReferences>();
+            nugetReferences.Setup(n => n.IsTransitivelyAvailableAsync(It.IsAny<string>(), default))
+                .Returns(new ValueTask<bool>(false));
+            var projectFile = mock.Mock<IProjectFile>();
+            projectFile.Setup(f => f.IsSdk).Returns(true);
+            projectFile.Setup(f => f.Sdk).Returns(new HashSet<string>(new[] { "Microsoft.NET.Sdk.Web" }));
```

Comment:
Are all these properties needed for your test? These look like they had been copied from a different test that needed different properties set. This should mock a minimal IProject